src/main/java/spoon/reflect/visitor/chain/ChainableFunction.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (C) 2006-2016 INRIA and contributors
+ * Spoon - http://spoon.gforge.inria.fr/
+ *
+ * This software is governed by the CeCILL-C License under French law and
+ * abiding by the rules of distribution of free software. You can use, modify
+ * and/or redistribute the software under the terms of the CeCILL-C license as
+ * circulated by CEA, CNRS and INRIA at http://www.cecill.info.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the CeCILL-C License for more details.
+ *
+ * The fact that you are presently reading this means that you have had
+ * knowledge of the CeCILL-C license and that you accept its terms.
+ */
+package spoon.reflect.visitor.chain;
+
+/**
+ * Functional interface used to map elements using a function which accepts one input parameter and calls output.accept(out) for each produced output element
+ *
+ * It is used for example by {@link QueryStep#map(ChainableFunction))}
+ *
+ * @param <T> the type of the input to the function
+ * @param <R> the type of the result produced by this function
+ */
+public interface ChainableFunction<T, R> {
+	void apply(T input, Consumer<R> output);

```

Comment:
Looking at Scann, basically Consumer is the implicit Function. What if Consumer extends Function (accept becomes apply) and then you only have one method map?